app/runnable.py

```diff
@@ -19,93 +21,119 @@
-            tokenized_answer.append(" ".join(split_answer))
-
+        tokenized_answer = self.get_tokenized_answer(input_data.user_answer)
         tokenized_answer_embedding = self.model.encode(tokenized_answer)
         similarity_scores = cosine_similarity(problem.embedded_keywords, tokenized_answer_embedding)
+
         predicts = []
+        keyword_score_dict = {standard.id: [None, 0, standard.content] for standard in input_data.keyword_standards}
         for keyword_idx, embedded_keyword_token_idx in enumerate(similarity_scores.argmax(axis=1)):
-            if self.threshold < similarity_scores[keyword_idx][embedded_keyword_token_idx]:
+            target_keyword_id = problem.keyword_standards[keyword_idx].id
+            score = similarity_scores[keyword_idx][embedded_keyword_token_idx]
+            if self.threshold < score and keyword_score_dict[target_keyword_id][1] < score:
+                keyword_score_dict[target_keyword_id][:2] = embedded_keyword_token_idx, score
+
+        for keyword_id, (embedded_keyword_token_idx, score, content) in keyword_score_dict.items():
+            if score > self.threshold:
                 start_idx = input_data.user_answer.find(tokenized_answer[embedded_keyword_token_idx])
                 end_idx = start_idx + len(tokenized_answer[embedded_keyword_token_idx])
                 predicts.append(
                     KeywordResponse(
-                        id=problem.keyword_standards[keyword_idx].id,
-                        keyword=problem.keyword_standards[keyword_idx].content,
+                        id=keyword_id,
+                        keyword=content,
                         predict_keyword_position=[start_idx, end_idx],
                         predict_keyword=input_data.user_answer[start_idx:end_idx],
                     )
                 )
-
-        return KeywordGradingResponse(problem_id=input_data.problem_id, correct_keywords=predicts)
+        response_data = KeywordGradingResponse(problem_id=input_data.problem_id, correct_keywords=predicts)
+        log.info(pformat(response_data.__dict__))
+        return response_data
 
 
 class ContentPredictRunnable(bentoml.Runnable):
-    SUPPORTED_RESOURCES = ("cpu",)
+
     SUPPORTS_CPU_MULTI_THREADING = False
 
     def __init__(self):
         self.model = get_content_grading_model()
         self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        log.info(f"content predict model is running on : {self.device}")
         self.template = self.model.template
         self.verbalizer = self.model.verbalizer
         special_tokens_dict = {"additional_special_tokens": ["</s>", "<unk>", "<pad>"]}
         self.model.tokenizer.add_special_tokens(special_tokens_dict)
         self.wrapped_tokenizer = T5TokenizerWrapper(
             max_seq_length=256, decoder_max_length=3, tokenizer=self.model.tokenizer, truncate_method="head"
         )
+        self.model = self.model.to(self.device)
```

Comment:
```suggestion
        content_grading_model = get_content_grading_model()
        self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
        log.info(f"content predict model is running on : {self.device}")
        self.template = content_grading_model.template
        self.verbalizer = content_grading_model.verbalizer
        special_tokens_dict = {"additional_special_tokens": ["</s>", "<unk>", "<pad>"]}
        content_grading_model.tokenizer.add_special_tokens(special_tokens_dict)
        self.wrapped_tokenizer = T5TokenizerWrapper(
            max_seq_length=256, decoder_max_length=3, tokenizer=content_grading_model.tokenizer, truncate_method="head"
        )
        self.model = content_grading_model.to(self.device)
```

위에서 초기화를 하고, 다시 세팅 후 초기화 하는 것 보다 제대로 세팅까지 완료한 후에 self.model에 할당하는게 낫지 않을까요?

Reply:
좋은 의견이네요! 반영할게요~